packages/@aws-cdk/aws-ec2/lib/nat.ts

```diff
@@ -45,6 +61,11 @@ export abstract class NatProvider {
    * Called by the VPC to configure NAT
    */
   public abstract configureNat(options: ConfigureNatOptions): void;
+
+  /**
```

Comment:
Maybe it's even better to add a method here, something like:

```
  public abstract configureSubnet(subnet: PrivateSubnet): void;
```

And put the code that's inside the "for" loop in that method. Then you can also just call that method to have routes applied to your subnets.

Reply:
👍